### PR TITLE
added 25_05 homework (submit form)

### PR DESCRIPTION
**Pull request by ArtstiomEzepchick**
Please check my homework, smilejke

### CODE REVIEW

File: 25_05/index.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            };
+
+            if (input.name === 'email') {
+                const isValid = checkEmail(event.target.value);
+                if (!isValid) {
+                    createError(input, 'has wrong format');
+                    return;
+                };
+
+                if (input.value.includes('.ru')) { // проверка на домен .ru
```
Comment (smilejke) [inline 25_05/index.js:77]: What if user will have `.ru` not as domain?
example: vadim.test.ru@gmail.com - seems to be valid email but your validation will check it as incorrect

File: 25_05/index.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    event.preventDefault();
+
+    for (let input of inputs) {
+        if (!input.value) {
+            createError(input, 'is empty');
+        };
+    };
+
+    if (termsCheckbox.checked) {
+        !Object.keys(errorsContainer).length ?
```
Comment (smilejke) [inline 25_05/index.js:138]: `Object.keys(errorsContainer)` or `Object.keys(errorsContainer).length` should be declared as a variable here

File: 25_05/index.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    });
+
+    input.addEventListener('blur', event => {
+        if (input.name !== 'fname' && input.name !== 'lname' && input.name !== 'terms') {
+            if (input.value.length <= 4) { // проверка на количество символов во всех инпутах, кроме fname, lname, terms
+                createError(input, 'must contain at least 5 characters');
+                return;
+            };
+
+            if (input.value.includes(' ')) { // проверка на пробелы во всех инпутах, кроме fname и lname
```
Comment (smilejke) [inline 25_05/index.js:65]: What if user will paste a space in the end?
example: 'Username '. It is not user-friendly to make it an error.
Check how `String.trim()` works or use regExp.

File: 25_05/index.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            if (input.name === 'pass') { // проверка на минимальные условия пароля
+                const isValid = checkPassword(event.target.value);
+                if (!isValid) {
+                    alert('Your password must contain at least 1 lowercase letter, 1 uppercase letter, 1 number, 1 symbol');
+                    createError(input, 'has wrong format');
+                    return;
+                };
+            };
+
+            if (input.name === 'tel' || input.name === 'postcode') { // проверка почты и телефона "только на цифры"
```
Comment (smilejke) [inline 25_05/index.js:92]: If there is no tel code field (like Telegram has two inputs: first is for code (country), second is for number), usually default `input type='tel'` behavior is to use "+" at the beginning, but due to your validation "+" is not a number, so tel with `+${code}${number}` format will fail your validation. It's not a mistake here, just a note. Will discuss it during our class.

**Review by smilejke — CHANGES_REQUESTED** [review]

File: 25_05/index.js (hunk 1 above)
Comment (ArtstiomEzepchick) [reply 25_05/index.js]: refactored

File: 25_05/index.js (hunk 2 above)
Comment (ArtstiomEzepchick) [reply 25_05/index.js]: done

File: 25_05/index.js (hunk 3 above)
Comment (ArtstiomEzepchick) [reply 25_05/index.js:69]: fixed

File: 25_05/index.js (hunk 4 above)
Comment (ArtstiomEzepchick) [reply 25_05/index.js:96]: got it

**Pull request merged**